### PR TITLE
upgrade log4j2 version and fix log4j2.xml provided sample

### PR DESCRIPTION
**Pull request by olamy**
add disruptor jar if users wants to use async logging

Signed-off-by: olivier lamy <oliver.lamy@gmail.com>

### CODE REVIEW

**Comment by olamy** [comment]
good idea! done

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 1]
```diff
@@ -265,4 +266,46 @@ public void testUnixSocket() throws Exception
...
+        DistributionTester distribution = DistributionTester.Builder.newInstance()
+            .jettyVersion(jettyVersion)
+            .jettyBase(jettyBase)
+            .mavenLocalRepository(System.getProperty("mavenRepoPath"))
+            .build();
+
+        String[] args1 = {
+            "--create-startd",
+            "--approve-all-licenses",
+            "--add-to-start=resources,server,http,webapp,deploy,jsp,servlet,servlets,logging-log4j2"
```
Comment (joakime) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: Can you test that the `${jetty.base}/resources/` has the log4j2.xml file?

Comment (olamy) [reply tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: for my curiosity what is the difference with testing that few lines later after the start?

Comment (olamy) [reply tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: @joakime got it fixed

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 2]
```diff
@@ -265,4 +266,47 @@ public void testUnixSocket() throws Exception
...
+        };
+        try (DistributionTester.Run run1 = distribution.start(args1))
+        {
+            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+            assertTrue(Paths.get(jettyBase.toString(), "resources", "log4j2.xml" ).toFile().exists());
```
Comment (joakime) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: Cool. this is what I had in mind.

One niggle.  why can't you use ...

``` java
assertTrue(Files.exists(jettyBase.resolve("resources/log4j2.xml")));
```

That would also be compatible with the File Systems present on Unix, OSX, and Windows automatically.
The less back and forth translation between `Path` and `String` the better.

**Comment by gregw** [comment]
@olamy could we get this merged before the release?   As @joakime  is out, I think it is OK to merge if you feel you have addressed all his review issues.

**Comment by olamy** [comment]
while testing this I found this issue https://github.com/eclipse/jetty.project/issues/3733
I should have a fix ready.

**Comment by olamy** [comment]
need #3733 first :)

**Pull request merged**